From mboxrd@z Thu Jan  1 00:00:00 1970
Received: from NAM04-DM6-obe.outbound.protection.outlook.com (mail-dm6nam04on2082.outbound.protection.outlook.com [40.107.102.82])
	(using TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384 (256/256 bits))
	(No client certificate requested)
	by smtp.subspace.kernel.org (Postfix) with ESMTPS id 6405379F2
	for <linux-input@vger.kernel.org>; Mon, 23 Jun 2025 08:55:26 +0000 (UTC)
Authentication-Results: smtp.subspace.kernel.org; arc=fail smtp.client-ip=40.107.102.82
ARC-Seal:i=2; a=rsa-sha256; d=subspace.kernel.org; s=arc-20240116;
	t=1750668927; cv=fail; b=KlTrx4iHXJ9ds+rYnSIw9cZHuZtfzcP7Psu3yFTeFlmzVks6a4pfLoRibT3X/8xHNhFdXcNmjiSwmEEaff3ZpWj+9OPnnOCBOIoab/L4ltQnlOPI+JUL7oh9vD99V301fw0+PoFaXZG6BPbZzd6dTo0z86Mv8e0TUS/HDw+D8OE=
ARC-Message-Signature:i=2; a=rsa-sha256; d=subspace.kernel.org;
	s=arc-20240116; t=1750668927; c=relaxed/simple;
	bh=SyGhHvcijcB424mhfv96r3vl3B9QO3Oz5T1fifjP+rQ=;
	h=From:To:CC:Subject:Date:Message-ID:MIME-Version:Content-Type; b=mqAeH3SVdoIrXpzpTzAWESEOXFLi6CW3ku7HB8Z/xDEVXomQ/FbHfTR4FcLOQYg9+c7FDxF586sFFeHbkKr/Rcvs+MFznoxGcZtiPxa4rU0GNxxTP/I0Nk1dS1pqMuj9sE2TvgldjpVDUtWiM0M7IdDDWb55FP6Im2igqn+XWBk=
ARC-Authentication-Results:i=2; smtp.subspace.kernel.org; dmarc=pass (p=quarantine dis=none) header.from=amd.com; spf=fail smtp.mailfrom=amd.com; dkim=pass (1024-bit key) header.d=amd.com header.i=@amd.com header.b=371lZhq5; arc=fail smtp.client-ip=40.107.102.82
Authentication-Results: smtp.subspace.kernel.org; dmarc=pass (p=quarantine dis=none) header.from=amd.com
Authentication-Results: smtp.subspace.kernel.org; spf=fail smtp.mailfrom=amd.com
Authentication-Results: smtp.subspace.kernel.org;
	dkim=pass (1024-bit key) header.d=amd.com header.i=@amd.com header.b="371lZhq5"
ARC-Seal: i=1; a=rsa-sha256; s=arcselector10001; d=microsoft.com; cv=none;
 b=u/tKw7ow6riB9Z4Z89Vb1ROZmnRNHU1AUco/Bjv8u3zYePGssSYnrG18MNai/M9/GPG8WcoLnEeIoGEkB8SOZkiNkJmZXwqPV1g1pBW6SOB6NDa3qPE8sj4blJAeml5HLSA5GZ6QuLDQqWxJmhcNRjeUJkvFoguigg4XR3lRXRVuHiUoY0D54HBrbTzWdKWYR2IeDtE4UCOtF5qiVPuYZTrwBRfZfzFDfgeL7QtLy5GuaW6Xp4eCfjmH2gzYdSKZaE8ZQrXowDZR/2UT1j5fV820Z4W9BwdEVEJv7VmcPJEavy6QFMKmfsultSnKusVUF2TuiO7gzh3X4tvQLODNuw==
ARC-Message-Signature: i=1; a=rsa-sha256; c=relaxed/relaxed; d=microsoft.com;
 s=arcselector10001;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-AntiSpam-MessageData-ChunkCount:X-MS-Exchange-AntiSpam-MessageData-0:X-MS-Exchange-AntiSpam-MessageData-1;
 bh=SjvvdakTUlwozCtMGPpMf0BHCWfWCcIt+H3IdEFiO+o=;
 b=Kz0hdE2CHsDHZ1qP+kOyXLMwGkYzcNHnnjXw8O9npNcANR8IigUK3UlgO1ZTG5m/9SVO8iRmIHNi4j7xj393oxLLFqoAenmYMfqtBAj6EeZ/LA5KVHSc0PqRJPjO/LL1+9/sBWhzfY/42jJSWIrTkfIIlnXDIowz1ABi26jdbeoh+NNjHrDCGhXFWB753mi2yfdruE8/M88oTofeCLDmlFVhg3qBwdHXBOV4J4nraAroNZ8kmwCjExDPqqaf8tyIZ8y5t/ZHQtSUtpAGbttZPlwMt4Piqfb5LMNud0sfaBi/B5S1EVkQc/CShS14H+J7aT0RV2VYM0SFEPoTnT4tsg==
ARC-Authentication-Results: i=1; mx.microsoft.com 1; spf=pass (sender ip is
 165.204.84.17) smtp.rcpttodomain=kernel.org smtp.mailfrom=amd.com; dmarc=pass
 (p=quarantine sp=quarantine pct=100) action=none header.from=amd.com;
 dkim=none (message not signed); arc=none (0)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=amd.com; s=selector1;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-SenderADCheck;
 bh=SjvvdakTUlwozCtMGPpMf0BHCWfWCcIt+H3IdEFiO+o=;
 b=371lZhq53bJtp/SdE93+HiDOZwfo+NaTvdJSVfRybjVr2V01lQE1lSfUUmGec2G6xmEUeSoddo/y/rXf8UIUlWKo9SBzK3hnx33LKw85B+I/W2SFAOAsh3FIKM4qR8AEzZNl5wXtiLvz68WKcGjWZ/Sfxv9pNX0h3P/kMlYVbn8=
Received: from BN9P222CA0002.NAMP222.PROD.OUTLOOK.COM (2603:10b6:408:10c::7)
 by DS0PR12MB7510.namprd12.prod.outlook.com (2603:10b6:8:132::20) with
 Microsoft SMTP Server (version=TLS1_2,
 cipher=TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384) id 15.20.8857.28; Mon, 23 Jun
 2025 08:55:23 +0000
Received: from BN3PEPF0000B06C.namprd21.prod.outlook.com
 (2603:10b6:408:10c:cafe::9e) by BN9P222CA0002.outlook.office365.com
 (2603:10b6:408:10c::7) with Microsoft SMTP Server (version=TLS1_3,
 cipher=TLS_AES_256_GCM_SHA384) id 15.20.8857.30 via Frontend Transport; Mon,
 23 Jun 2025 08:55:22 +0000
X-MS-Exchange-Authentication-Results: spf=pass (sender IP is 165.204.84.17)
 smtp.mailfrom=amd.com; dkim=none (message not signed)
 header.d=none;dmarc=pass action=none header.from=amd.com;
Received-SPF: Pass (protection.outlook.com: domain of amd.com designates
 165.204.84.17 as permitted sender) receiver=protection.outlook.com;
 client-ip=165.204.84.17; helo=SATLEXMB04.amd.com; pr=C
Received: from SATLEXMB04.amd.com (165.204.84.17) by
 BN3PEPF0000B06C.mail.protection.outlook.com (10.167.243.71) with Microsoft
 SMTP Server (version=TLS1_2, cipher=TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256) id
 15.20.8901.1 via Frontend Transport; Mon, 23 Jun 2025 08:55:22 +0000
Received: from airavat.amd.com (10.180.168.240) by SATLEXMB04.amd.com
 (10.181.40.145) with Microsoft SMTP Server (version=TLS1_2,
 cipher=TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256) id 15.1.2507.39; Mon, 23 Jun
 2025 03:55:20 -0500
From: Basavaraj Natikar <Basavaraj.Natikar@amd.com>
To: <jikos@kernel.org>, <bentiss@kernel.org>
CC: <Mario.Limonciello@amd.com>, <linux-input@vger.kernel.org>, "Basavaraj
 Natikar" <Basavaraj.Natikar@amd.com>, Eric Naim <dnaim@cachyos.org>, "Akshata
 MukundShetty" <akshata.mukundshetty@amd.com>
Subject: [PATCH v2] HID: amd_sfh: Enable operating mode
Date: Mon, 23 Jun 2025 14:24:53 +0530
Message-ID: <20250623085453.2267034-1-Basavaraj.Natikar@amd.com>
X-Mailer: git-send-email 2.34.1
Precedence: bulk
X-Mailing-List: linux-input@vger.kernel.org
List-Id: <linux-input.vger.kernel.org>
List-Subscribe: <mailto:linux-input+subscribe@vger.kernel.org>
List-Unsubscribe: <mailto:linux-input+unsubscribe@vger.kernel.org>
MIME-Version: 1.0
Content-Transfer-Encoding: 8bit
Content-Type: text/plain
X-ClientProxiedBy: SATLEXMB04.amd.com (10.181.40.145) To SATLEXMB04.amd.com
 (10.181.40.145)
X-EOPAttributedMessage: 0
X-MS-PublicTrafficType: Email
X-MS-TrafficTypeDiagnostic: BN3PEPF0000B06C:EE_|DS0PR12MB7510:EE_
X-MS-Office365-Filtering-Correlation-Id: dd4ecc0d-2815-4819-7511-08ddb233b055
X-MS-Exchange-SenderADCheck: 1
X-MS-Exchange-AntiSpam-Relay: 0
X-Microsoft-Antispam:
	BCL:0;ARA:13230040|1800799024|376014|82310400026|36860700013;
X-Microsoft-Antispam-Message-Info:
	=?us-ascii?Q?C5uvoz0iTOXVwtEBj4bIsnAuyh7z3q+JGNHJUfuk89cQn678XoXLGD08UrVg?=
 =?us-ascii?Q?6Hyi50Zv1gzaa0viit5pNZyjl8A935VFW5J96/0MYBgbZ1Gsc2o7v3pBJslq?=
 =?us-ascii?Q?Bw7WKveDChu7DaHjlW+E47vVCAH1Q/GDO1OMHoNQaC1Je3O1sJ7jwNq2k+Ua?=
 =?us-ascii?Q?Lx3wJ38pIm3ymuALHlVipaBNECeW4V4dmi8WR22l/ubhZv757cjrMN9jsw5o?=
 =?us-ascii?Q?7gDb5Upk5DbFLtXLK+EU3mauxB0sMnKIoDfczpJ05uNzJk9/dIE1cN32PzJj?=
 =?us-ascii?Q?m/yvMcmM59NICxBe1seg1lU/Lo1dZveOXEB7wtq2ETdxRmAjnpbpwuhlAgQH?=
 =?us-ascii?Q?QN9+RGoy4RnAhl0z6Afw3xoNykAzQctZqaa/i5qDcpJM88Bjubv2/Kz7kRJA?=
 =?us-ascii?Q?5vVAi3Yfo+sr+jDNOtPINVg/7wNzBYkupVSPtRZNZQ9yuxZcTnlqu182mLlT?=
 =?us-ascii?Q?DbkSNzig/IsMEFKg0tFoEZebQs+q+hwTPqiDrcSXriIzLCuFDaZ7KQE9eVVV?=
 =?us-ascii?Q?g5CLnfgAKACYnxSwV+cUQTBFPZMz4RGp8HB2i1Z+QDx4QMZTE64ChReoGXb0?=
 =?us-ascii?Q?fCsdrE49HnKJEfbgOtXmq87SMoQV/PTsS8Bukf8r0gEfZ67Y0a5fzn1T3qZx?=
 =?us-ascii?Q?Srr6j1zt068sCMeUYF8FBhcsfEpDcyPOAG5xU4oMhX4uRzsBUDcvrTfmOMo2?=
 =?us-ascii?Q?aFqOWpQ9Jb8fi3VC5lHIi8ERNVkJz5e79H0IakeWTbRxmJ7hmkUV/rrxyzWH?=
 =?us-ascii?Q?88N7P1NsYxRG/qUD13W1BlGSCyr1zEUyPTg1VrLIdVvqxBpnQRWH7P+/W2D/?=
 =?us-ascii?Q?nDpAemcFd7ryiwkiZwieoTHq/LAKmlrYpHSMFbB2EQk2zi0HnfR9epYlzzlD?=
 =?us-ascii?Q?6mWwUOtHRq/m9pN5DEYSQefPtvAZ1rJiKH2JeZ2mrO/e+LN0nEr5ZbEuEB9W?=
 =?us-ascii?Q?lixsUl3PGG3rdAq+djBd5DhJFcHel0zdRTn3ZIi95GK53r60ZCtPKwTo3gFU?=
 =?us-ascii?Q?Z1kF7EJl2OAEv4obDgLLSYS8PBFNvWEOuR6SbFo9v3+ZU8NsSjksIhAQsq7J?=
 =?us-ascii?Q?568IRlXWgSUtWuFIwch9USVEMCOd3lxDRtaT8eQCxgd/nmai7jjmafU+vRri?=
 =?us-ascii?Q?Zyuc9wHVonGghE2GgnWSmu8H2AvYts9Wwd3Vd09FozCLdsulSELPlTbh/Mj2?=
 =?us-ascii?Q?Uoce2Ln2jAxsPby7f46kqjjhmltgHo89SsQWQo/AYliG7mWjmQwlc8gZXxKV?=
 =?us-ascii?Q?R0Hltc1LnQPFdh63wYbHOeAqmivEcMbcaxjAVBHEh1VA4d8IuodWdOEc3Nh6?=
 =?us-ascii?Q?iOpK+myW+BAoDt8prI0u+wLUXlMbQUZ4bHHA6aoONvorzT8I+c0ijUD8+G1/?=
 =?us-ascii?Q?mxNguzQQ3A5Qglwsi8wcr2+Om1mGcmD4ePOD5HVvFesKTAjJkrarDd/ADvCI?=
 =?us-ascii?Q?S5/85f7FRwUHDYu3dW2ujci79EGgjf5F3CEm8Iz+cAwNowhP2ab90WHRU7p5?=
 =?us-ascii?Q?K1Wr02cxijYNNwS+yh6JNU3tUOtQyTFipvE7?=
X-Forefront-Antispam-Report:
	CIP:165.204.84.17;CTRY:US;LANG:en;SCL:1;SRV:;IPV:CAL;SFV:NSPM;H:SATLEXMB04.amd.com;PTR:InfoDomainNonexistent;CAT:NONE;SFS:(13230040)(1800799024)(376014)(82310400026)(36860700013);DIR:OUT;SFP:1101;
X-OriginatorOrg: amd.com
X-MS-Exchange-CrossTenant-OriginalArrivalTime: 23 Jun 2025 08:55:22.7526
 (UTC)
X-MS-Exchange-CrossTenant-Network-Message-Id: dd4ecc0d-2815-4819-7511-08ddb233b055
X-MS-Exchange-CrossTenant-Id: 3dd8961f-e488-4e60-8e11-a82d994e183d
X-MS-Exchange-CrossTenant-OriginalAttributedTenantConnectingIp: TenantId=3dd8961f-e488-4e60-8e11-a82d994e183d;Ip=[165.204.84.17];Helo=[SATLEXMB04.amd.com]
X-MS-Exchange-CrossTenant-AuthSource:
	BN3PEPF0000B06C.namprd21.prod.outlook.com
X-MS-Exchange-CrossTenant-AuthAs: Anonymous
X-MS-Exchange-CrossTenant-FromEntityHeader: HybridOnPrem
X-MS-Exchange-Transport-CrossTenantHeadersStamped: DS0PR12MB7510

Add changes to enable operating modes in the driver to allow the FW to
activate and retrieve data from relevant sensors. This enables the FW to
take necessary actions based on the operating modes.

Tested-by: Eric Naim <dnaim@cachyos.org>
Co-developed-by: Akshata MukundShetty <akshata.mukundshetty@amd.com>
Signed-off-by: Akshata MukundShetty <akshata.mukundshetty@amd.com>
Signed-off-by: Basavaraj Natikar <Basavaraj.Natikar@amd.com>
---
v2: 
* Increased missing device count.
---
 drivers/hid/amd-sfh-hid/amd_sfh_client.c | 23 +++++++++++++++++++++++
 drivers/hid/amd-sfh-hid/amd_sfh_hid.h    |  2 +-
 drivers/hid/amd-sfh-hid/amd_sfh_pcie.c   |  4 ++++
 drivers/hid/amd-sfh-hid/amd_sfh_pcie.h   |  1 +
 4 files changed, 29 insertions(+), 1 deletion(-)

diff --git a/drivers/hid/amd-sfh-hid/amd_sfh_client.c b/drivers/hid/amd-sfh-hid/amd_sfh_client.c
index 3438d392920f..0f2cbae39b2b 100644
--- a/drivers/hid/amd-sfh-hid/amd_sfh_client.c
+++ b/drivers/hid/amd-sfh-hid/amd_sfh_client.c
@@ -146,6 +146,8 @@ static const char *get_sensor_name(int idx)
 		return "gyroscope";
 	case mag_idx:
 		return "magnetometer";
+	case op_idx:
+		return "operating-mode";
 	case als_idx:
 	case ACS_IDX: /* ambient color sensor */
 		return "ALS";
@@ -243,6 +245,20 @@ int amd_sfh_hid_client_init(struct amd_mp2_dev *privdata)
 			rc = -ENOMEM;
 			goto cleanup;
 		}
+
+		if (cl_data->sensor_idx[i] == op_idx) {
+			info.period = AMD_SFH_IDLE_LOOP;
+			info.sensor_idx = cl_data->sensor_idx[i];
+			info.dma_address = cl_data->sensor_dma_addr[i];
+			mp2_ops->start(privdata, info);
+			cl_data->sensor_sts[i] = amd_sfh_wait_for_response(privdata,
+									   cl_data->sensor_idx[i],
+									   SENSOR_ENABLED);
+			if (cl_data->sensor_sts[i] == SENSOR_ENABLED)
+				cl_data->is_any_sensor_enabled = true;
+			continue;
+		}
+
 		cl_data->sensor_sts[i] = SENSOR_DISABLED;
 		cl_data->sensor_requested_cnt[i] = 0;
 		cl_data->cur_hid_dev = i;
@@ -303,6 +319,13 @@ int amd_sfh_hid_client_init(struct amd_mp2_dev *privdata)
 
 	for (i = 0; i < cl_data->num_hid_devices; i++) {
 		cl_data->cur_hid_dev = i;
+		if (cl_data->sensor_idx[i] == op_idx) {
+			dev_dbg(dev, "sid 0x%x (%s) status 0x%x\n",
+				cl_data->sensor_idx[i], get_sensor_name(cl_data->sensor_idx[i]),
+				cl_data->sensor_sts[i]);
+			continue;
+		}
+
 		if (cl_data->sensor_sts[i] == SENSOR_ENABLED) {
 			rc = amdtp_hid_probe(i, cl_data);
 			if (rc)
diff --git a/drivers/hid/amd-sfh-hid/amd_sfh_hid.h b/drivers/hid/amd-sfh-hid/amd_sfh_hid.h
index 1c91be8daedd..7452b0302953 100644
--- a/drivers/hid/amd-sfh-hid/amd_sfh_hid.h
+++ b/drivers/hid/amd-sfh-hid/amd_sfh_hid.h
@@ -11,7 +11,7 @@
 #ifndef AMDSFH_HID_H
 #define AMDSFH_HID_H
 
-#define MAX_HID_DEVICES		6
+#define MAX_HID_DEVICES		7
 #define AMD_SFH_HID_VENDOR	0x1022
 #define AMD_SFH_HID_PRODUCT	0x0001
 
diff --git a/drivers/hid/amd-sfh-hid/amd_sfh_pcie.c b/drivers/hid/amd-sfh-hid/amd_sfh_pcie.c
index 1c1fd63330c9..2983af969579 100644
--- a/drivers/hid/amd-sfh-hid/amd_sfh_pcie.c
+++ b/drivers/hid/amd-sfh-hid/amd_sfh_pcie.c
@@ -29,6 +29,7 @@
 #define ACEL_EN		BIT(0)
 #define GYRO_EN		BIT(1)
 #define MAGNO_EN	BIT(2)
+#define OP_EN		BIT(15)
 #define HPD_EN		BIT(16)
 #define ALS_EN		BIT(19)
 #define ACS_EN		BIT(22)
@@ -232,6 +233,9 @@ int amd_mp2_get_sensor_num(struct amd_mp2_dev *privdata, u8 *sensor_id)
 	if (MAGNO_EN & activestatus)
 		sensor_id[num_of_sensors++] = mag_idx;
 
+	if (OP_EN & activestatus)
+		sensor_id[num_of_sensors++] = op_idx;
+
 	if (ALS_EN & activestatus)
 		sensor_id[num_of_sensors++] = als_idx;
 
diff --git a/drivers/hid/amd-sfh-hid/amd_sfh_pcie.h b/drivers/hid/amd-sfh-hid/amd_sfh_pcie.h
index 05e400a4a83e..2eb61f4e8434 100644
--- a/drivers/hid/amd-sfh-hid/amd_sfh_pcie.h
+++ b/drivers/hid/amd-sfh-hid/amd_sfh_pcie.h
@@ -79,6 +79,7 @@ enum sensor_idx {
 	accel_idx = 0,
 	gyro_idx = 1,
 	mag_idx = 2,
+	op_idx = 15,
 	als_idx = 19
 };
 
-- 
2.34.1


From mboxrd@z Thu Jan  1 00:00:00 1970
Received: from NAM02-BN1-obe.outbound.protection.outlook.com (mail-bn1nam02on2058.outbound.protection.outlook.com [40.107.212.58])
	(using TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384 (256/256 bits))
	(No client certificate requested)
	by smtp.subspace.kernel.org (Postfix) with ESMTPS id DDB9D143748
	for <linux-input@vger.kernel.org>; Mon, 23 Jun 2025 14:59:22 +0000 (UTC)
Authentication-Results: smtp.subspace.kernel.org; arc=fail smtp.client-ip=40.107.212.58
ARC-Seal:i=2; a=rsa-sha256; d=subspace.kernel.org; s=arc-20240116;
	t=1750690764; cv=fail; b=Q+2XekYDkF2bYkr0P/6F/ro/iKfvlWinMLDTWBlgDb/gcdV3JgymZ25X6AGqeoo2xVx7rL+32ioQs9n+wN3JYrSGNf9pJjH8rqP3HykUq5hpoJY/HbvQDEtb+4dK6ifof9kHFZf1vwPNCGDd13wRL1Pta5T2hoR3ZuZ22MkfSAs=
ARC-Message-Signature:i=2; a=rsa-sha256; d=subspace.kernel.org;
	s=arc-20240116; t=1750690764; c=relaxed/simple;
	bh=zqjPJ+Nn9Umqyva1WNm6gu3R+cnkFlKrKV0KBk2ASIM=;
	h=From:To:CC:Subject:Date:Message-ID:References:In-Reply-To:
	 Content-Type:MIME-Version; b=SfKiA5fn7OdXkcNFQfydSpyLjw1UrOUM7jssXdeE4+6P8/UM5Lj43FRMHIagaENFrIJv6CE6sBWs3MCLK8W4rUn9raoxnaRve8E0N5YdCPUVrVmXsHaxpPgn+4w347HP45K4gN78GOwebBMkFvMfvirLaQIp+HSMesWIJz1HWe0=
ARC-Authentication-Results:i=2; smtp.subspace.kernel.org; dmarc=pass (p=quarantine dis=none) header.from=amd.com; spf=fail smtp.mailfrom=amd.com; dkim=pass (1024-bit key) header.d=amd.com header.i=@amd.com header.b=ByRiO+Sa; arc=fail smtp.client-ip=40.107.212.58
Authentication-Results: smtp.subspace.kernel.org; dmarc=pass (p=quarantine dis=none) header.from=amd.com
Authentication-Results: smtp.subspace.kernel.org; spf=fail smtp.mailfrom=amd.com
Authentication-Results: smtp.subspace.kernel.org;
	dkim=pass (1024-bit key) header.d=amd.com header.i=@amd.com header.b="ByRiO+Sa"
ARC-Seal: i=1; a=rsa-sha256; s=arcselector10001; d=microsoft.com; cv=none;
 b=UdHMV/ajOjWbz+YuA1ltXV1yZEdD5dOWaHbLTd23MbR0ZdsH+j271ZUHmZAb0ysIYbcqsEMNwM4q01wIpSe0bvSFFmdNQQkfIg8ghkPukNsmJgjFF8q2Tgf5c1DqfWdIaZd4Meo+mnXh7MfCJB4Mun/7B1EkXKUjVSqA00onJoILxdQXAbPRhjv3iNDp+EuwboQ5Eq+1SalaLFQsReHDuTEck1VVyO5suQ9nD+UncYPvofcubhHBvq8dY6LCrriwN/yIBg6Kd9PxojxRJ7gRUUXxUGyrrWYSBVVbHY6Cbz0QWmHj4FEjqB5X3ovnFPgEzVKtxkLGAFaUKrPGJSIW8A==
ARC-Message-Signature: i=1; a=rsa-sha256; c=relaxed/relaxed; d=microsoft.com;
 s=arcselector10001;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-AntiSpam-MessageData-ChunkCount:X-MS-Exchange-AntiSpam-MessageData-0:X-MS-Exchange-AntiSpam-MessageData-1;
 bh=zqjPJ+Nn9Umqyva1WNm6gu3R+cnkFlKrKV0KBk2ASIM=;
 b=nOMidKo7ktnzQW90o1poyP3wFWA+mVBxpUvUh4R9qkX8O3j9aaf08uwxFEE05G+o1SSRcHoOQE1FlBtqvF/MnsR0wXMBy5jBvZ/Grey0pIhQjog3uOzclEz9wPKQwREz1t6MyzCy3Oi2Qv1J3uW40OFu1DDRtIX86b1kpA3PjzYg3gx9XQXS8fp2izpH9yNCyUdTHtNLEQWPVH/tYXrEZLalPvILmr1qLSvD9504EstD8LTwh57MqdSVQcmohrKWy5RlZvP6pfdfQ5S72JJ+Q5Q1o2DpJ1JU3YUxQE+gJVB5hAGJwYWpx+cuM2b9vRx4DBix5ehJE04Yb8C5zMG7eg==
ARC-Authentication-Results: i=1; mx.microsoft.com 1; spf=pass
 smtp.mailfrom=amd.com; dmarc=pass action=none header.from=amd.com; dkim=pass
 header.d=amd.com; arc=none
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=amd.com; s=selector1;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-SenderADCheck;
 bh=zqjPJ+Nn9Umqyva1WNm6gu3R+cnkFlKrKV0KBk2ASIM=;
 b=ByRiO+SaPGegiE4rv4bAbzN8BpATvc+OVgwJQzvoggZa+JyxIcms7NWzlJ07wMCst7dejmPZhUc4Oe8dKE4GkWgoOtVi8BMyFWQotx22W6cpehCxsomzw434ca/TfyD/x5sPdDg66f6Ixcqn745XtRA4uYovngROEiipjXFsGx4=
Received: from MN0PR12MB6101.namprd12.prod.outlook.com (2603:10b6:208:3cb::10)
 by CH8PR12MB9766.namprd12.prod.outlook.com (2603:10b6:610:2b6::10) with
 Microsoft SMTP Server (version=TLS1_2,
 cipher=TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384) id 15.20.8835.28; Mon, 23 Jun
 2025 14:59:20 +0000
Received: from MN0PR12MB6101.namprd12.prod.outlook.com
 ([fe80::37ee:a763:6d04:81ca]) by MN0PR12MB6101.namprd12.prod.outlook.com
 ([fe80::37ee:a763:6d04:81ca%7]) with mapi id 15.20.8857.026; Mon, 23 Jun 2025
 14:59:20 +0000
From: "Limonciello, Mario" <Mario.Limonciello@amd.com>
To: "Natikar, Basavaraj" <Basavaraj.Natikar@amd.com>, "jikos@kernel.org"
	<jikos@kernel.org>, "bentiss@kernel.org" <bentiss@kernel.org>
CC: "linux-input@vger.kernel.org" <linux-input@vger.kernel.org>, Eric Naim
	<dnaim@cachyos.org>, "MukundShetty, Akshata" <Akshata.MukundShetty@amd.com>
Subject: Re: [PATCH v2] HID: amd_sfh: Enable operating mode
Thread-Topic: [PATCH v2] HID: amd_sfh: Enable operating mode
Thread-Index: AQHb5ByQAFLAhMOLzkC4yI+rp6nUXLQQ1pIA
Date: Mon, 23 Jun 2025 14:59:20 +0000
Message-ID: <329541ea-d047-4f8f-8e65-286f35603381@amd.com>
References: <20250623085453.2267034-1-Basavaraj.Natikar@amd.com>
In-Reply-To: <20250623085453.2267034-1-Basavaraj.Natikar@amd.com>
Accept-Language: en-US
Content-Language: en-US
X-MS-Has-Attach:
X-MS-TNEF-Correlator:
user-agent: Mozilla Thunderbird
authentication-results: dkim=none (message not signed)
 header.d=none;dmarc=none action=none header.from=amd.com;
x-ms-publictraffictype: Email
x-ms-traffictypediagnostic: MN0PR12MB6101:EE_|CH8PR12MB9766:EE_
x-ms-office365-filtering-correlation-id: 03a38bb2-decd-48e3-d336-08ddb266889e
x-ms-exchange-senderadcheck: 1
x-ms-exchange-antispam-relay: 0
x-microsoft-antispam: BCL:0;ARA:13230040|376014|366016|1800799024|38070700018;
x-microsoft-antispam-message-info:
 =?utf-8?B?ZVp3Um1RWW13bmdQR1V4N1lvc29TUWJEN2lKRkhxU2pUOUEzMEljMUR5eXVH?=
 =?utf-8?B?OTNUNGV0blF5STZBZkVKN0NiMjBob2JZRGRGWXFtTjhvZWZvUGxwMExhTGNL?=
 =?utf-8?B?SHU1Tm9zVTV3Z21lM0pXNVpoWEV5cnI5WVZ0Zm1TaHZiVHJHdnFCbmxtUWJ3?=
 =?utf-8?B?U3JDRkwxVDlmMTNyQ09EYnFQR2hEVlBpT0wyT0RhVjloazRraW1IWE9rWXNW?=
 =?utf-8?B?OFQwVGZEWE9sUDVKQlJjbDAycVJ4MlJEZkhUb3ZYR2xSbUM0eVJUVysrRlFv?=
 =?utf-8?B?aW8vK0ltSTRydTZzY0xCcDRlQWpycmQ0WXZYak5vdVhWVFVLUFJDc1BndFha?=
 =?utf-8?B?T2xIU2QrVHlPU1pzUzRMcjFYRGFjdzNrWk1FTVhPNXhidHJpZGlGUXhiT1lx?=
 =?utf-8?B?aVVoVytkdVp1bWRXLzIvSEZqUi9wMnV3STVEQVNWby9MVkhpVWNOd3pyaDJC?=
 =?utf-8?B?NWowYy9VaGJQOTRjdHNoU1QyaVIwRUltVjBzVm1ndFpybjZnNStHSWJRSlRC?=
 =?utf-8?B?bkgxNW5HL2UrdVB0WGgva1NJTDVCWW9sSGFMR3cxOExPeGF2TW1NSTllU045?=
 =?utf-8?B?WWM1Snl3WFhBeExFQnZzem1DMGZIVlltSGVzaWhOVnNVM2dkOE8rYTAxMGdX?=
 =?utf-8?B?Y3VQMGxDRzNhc2poeUFCWldrOGU3Y2Q0YnFuY1ZjUWQxWGl5WmVIYjNsME5O?=
 =?utf-8?B?a01wblBIRmE5c2NyME9MRUcxTHhUaHFmSG1BcW5SVEVZWEUvN24vc2liZFkr?=
 =?utf-8?B?bWQ0cnB5My9SWEo0ZUtEaG8rbnJhUDZRZHdqS1lpaWxsNWw4Qi9QT21CaDEv?=
 =?utf-8?B?NVJ0d0pPdVBFdFRJVDlDdWFHcE5UdG16azFkeFFoYTlKNGtQMkltUXpUeTc0?=
 =?utf-8?B?MVYrU0tHcDlNaGZ1ek5ENEtHbTFVcVVHZnFpN2tHZTVQekdhcVVvM1I3SThI?=
 =?utf-8?B?cWJuYm1hem9KSDJhcUlsdWU4T0xPOGI2VVY2aVBlYU53eXUzQWwyOERHV3lV?=
 =?utf-8?B?ajRlU2d0dDhYaHNueDVwc3U1K21zUGNSQVUxZEtBZS8xOFFiR2s1MjdDUjhI?=
 =?utf-8?B?Y25Zd21LVTA1eGlLMGVYYkFSVjBEOWVtMEJTMVpBRklLVjcydjg2aERlQi9S?=
 =?utf-8?B?aTBhMkl6b0ZjYXVYc0FQVWlrTzF0RnF2UG5nbUl6dFJWTWxKN1NQeXRlL1dB?=
 =?utf-8?B?akR0dDFDTFpaNnRVa010TDBudkZueEM0alVkdDZPNUhHUkdHUENoaWJBR2w4?=
 =?utf-8?B?SXB0K3NncnYwRUFmbWpJaXdpRTJ4QXhjS1o5NDY4SXhTK3BsL05oWk93RDk1?=
 =?utf-8?B?Q2xGTVc5TTI5eThPTEM5OEczR3lVamtZV3hhR0RaVUJ1MUVWRjlwZ2FtSWNl?=
 =?utf-8?B?Mk9Oc1VHOHk4dDlrcHNxeVNESXZlNXQrUzdqQmNGUmt1QzN3akJuM3VOcWNv?=
 =?utf-8?B?c0x1RUs0aWRKRlV0ME1SVHNrbExLaEhuMEdjSnpZeW9FWG5aMTRIa1RqTlBt?=
 =?utf-8?B?bTZzVGhQQk85a0dFWHpXZUdrbS9wWUF6TFhRK044S2lFOXhraVNva0xZZWIv?=
 =?utf-8?B?d1l1T2M0QmZ3TnB0cmhoUTMwdW1EVEFmaWFuVy8xc2s2MUtOdGFNcDBRR0cx?=
 =?utf-8?B?K3lyT1Bhd2NaMUFNR1plb3RHSWMyVHVkRGVvT0ZKYjdodGdaWmVLTTl2MlM0?=
 =?utf-8?B?eGg5dFR6RUwrNTh2WElIcEhxd0IyZVUzc0lFOGZhTmRyWHpDVFBjWVFIR20v?=
 =?utf-8?B?Wk9LTnBhT3JPODFlOEw5bEJyVlVqK3llUGdSc2dxQVRyUTBURk92L2hJbmlU?=
 =?utf-8?B?OEJua21WRG4xL3ZHeVNWZlM4dlA4Y05sOTRrZ0diM1BCTHl2dFhIYVVZUUpw?=
 =?utf-8?B?VXBkU0pMczgvSndHVnZkOUx0OHV0VVlkWEFTVjl3dVd1Ylh6Y3Awd3VvMlRJ?=
 =?utf-8?B?YURFaXl6dVVtYkUwZ2xQTzVZUDhVN1BNUzBCSEhEendlcU80elRrU1VFNTV6?=
 =?utf-8?Q?xN25AUKRRFB4hFCzNLBs7DQLIjPgDs=3D?=
x-forefront-antispam-report:
 CIP:255.255.255.255;CTRY:;LANG:en;SCL:1;SRV:;IPV:NLI;SFV:NSPM;H:MN0PR12MB6101.namprd12.prod.outlook.com;PTR:;CAT:NONE;SFS:(13230040)(376014)(366016)(1800799024)(38070700018);DIR:OUT;SFP:1101;
x-ms-exchange-antispam-messagedata-chunkcount: 1
x-ms-exchange-antispam-messagedata-0:
 =?utf-8?B?Q2ppNGp0M0cxdk1YZlA4WFdvNmgwbFRyc2FwRG5zQ2FyczJqdFhWek1JbHNH?=
 =?utf-8?B?V0Q0MUtZQllwT1pocGlHdHFaaVZLQmhUSjRzN1IzWDBiblNONXVBZThuQTZy?=
 =?utf-8?B?ZnZERCttamt0QisweXpaV0l5Skh2L2RYUnhxNXVNZVhSYSszdmRuTUZVSzJq?=
 =?utf-8?B?OGpXWjVNWDFTZ1RhMkUwN1pIVmp0OVpaWExZSE01cE8xOWc0L3ozOEYwQ0RP?=
 =?utf-8?B?UDl0b2tucGZBZklIVjVoVkoybVQxVEFvYkQ1OWFNT1pxK0lLS0xhdmVWRTBH?=
 =?utf-8?B?MENJdit6QlpzQ1ArMEJpZUlXdTVSKzM0MFVzVjEwRmhQUWFvbzIzT0poV3RM?=
 =?utf-8?B?TktZMkN5L0x5YVhmZk5jV1lIQ3RIcmxnMTloR253UDROUUNZV05JeHBOc2Jq?=
 =?utf-8?B?eStTYTkrN2V5cHpTS0pKZ04zdDE1SFdvZVFEV2hOVzcrWlg1Z1orTCtQeEtO?=
 =?utf-8?B?Qy9TTjRHemZDTzZSTUJ5RXpDMFBFQWRwV250SFZsQitCbXJGSzFNSTdjUjJw?=
 =?utf-8?B?eGg1bDBHbWs2bExWekl0N21LM2t5aTltaGRrY1ZzaEpFeGtRRU1iZzY3YnhV?=
 =?utf-8?B?UUFGYklpVWJLMXErL1B0eC9CT3lkTmgvV0Q4Y1BURXE0R1ROQ2VXZkZiRDI3?=
 =?utf-8?B?MlprZkpzWXN1eDBKK2JnS0VRV1VVK2kzV1NhYmhKcTNoYm43a2lmZTY2RFYr?=
 =?utf-8?B?aWRkMVl0TDIxYXh5MWRocDhZdnJ2VDFmNEo4WFpoSTBMYTUxZUdrZzBDWTRm?=
 =?utf-8?B?VC9QRXZKc1NtWENRTVY1T0ViMyt4UUZzNk5kUi8vbDBCR1F6V3h3aE9aNzFJ?=
 =?utf-8?B?bURVNThiR213aDZldmI1WVZkTk9nMFIzMFE0R2cwMXBZbC9TSWhNa0tlK0V4?=
 =?utf-8?B?SkV4cVloWG9MNkxzQi9KY2ZaNmFDUWNLck5NSllZUkNXWXdUNHliS1M3MUtV?=
 =?utf-8?B?TjBEaXdDekpZMkcvTmFGbjdFaVAwdUNoRGIyVmpadzQ5MjVxakQ3dXRCcVJY?=
 =?utf-8?B?VFRWYzA4cSs2Yzk5bGhMVlhVZS92Wm1hTnB6d0x6RXpIYStxOENVV3U3TjMy?=
 =?utf-8?B?a1NRYWdrTXJsUk02RC9KNVFwWnhyTTFsQVJGanRKaG1HSG95enVZR0xVVjNS?=
 =?utf-8?B?UEFheDdHSzc4eTI1SC9COEllWFNNMTVpRWNBZXN3b2wwdDN3b1pIWHZScHVB?=
 =?utf-8?B?cDBHeVdib3VLUVZ4K2FDR2gvQ21ybVVIWGxJTndzU01iRms1UlhHNXJ4UW1n?=
 =?utf-8?B?WlNENFk0aDM1RjF1VEZRNGYwT2RBa2hBenE4enJ0Z3pzVDltQWQ4SjFxMER5?=
 =?utf-8?B?VHhIZjF3WUlENTE3ODJkZmN1OE5xM3hFNmJ1ZjVOTlhmYnJuN2NKZTBGVnAx?=
 =?utf-8?B?MmV4UzJIbTNpdmY3VU11aDR2TnBteHhuQUdhZmF2SGFJdEpoTEo0WCt3OGZ0?=
 =?utf-8?B?cm1jUXlqSkp4ZmpTNFd6amdZZ3VuN3NXdFdXeW5jMFRjMWlvdVpTQVFQWEJt?=
 =?utf-8?B?Unk0ZmJMQkR3dGNhTjU2NWI2eUZ0Z3pNUjBhbHdQN1VWK1dWUWNjRjhVY0xV?=
 =?utf-8?B?TFZteHI3OHNaVlRCaXBDZngxUjZMSHJ2ZjVMV1NzN0pKTE9VVnYyMk1aVFdB?=
 =?utf-8?B?YkUwMGdxWk12RXRwTVJHRjl4RWY5MXBRcGxIY3FCRndrOENXdVpqVXRIRnhq?=
 =?utf-8?B?c1FIRVBBeWxQMUlMZFVFb21SbTY2Z21IbXh2UXhRTWZoR1RzbjFvOEF2a1RS?=
 =?utf-8?B?d1ZEaCttelcybkFrb2xJY0l3RHB5a1NFKy9yVXB4VEl0TEZSbFNIRFhpcEZy?=
 =?utf-8?B?YUJmd1o4bkM0UEJxRW9LTG5HR3kyMVpJTkxpcU50Z0s0VlRiVW10K1hCZHBF?=
 =?utf-8?B?SFFUTkVaWG9ETnBBUUVqZkRSY2cxaWdKM1g0cGMrbG52Vm5JY0JiMUk5OXd6?=
 =?utf-8?B?L1g1NDdLVE4wMXA4L1NPY1MxMVNTbzBhMDVxdzlKc1FZdXN0RUpPaXljRDBx?=
 =?utf-8?B?a2xQbXQ5bjVYanlQY083YUozdmc1Q1FCRVh5eEdyUUpOUnlUNHlJWVhiNUFO?=
 =?utf-8?B?WHE0RkxGODUwYmVETVdaMmg1M1FBa3gvZ2d6UFI4cU9PdDJ0R1ZycER4ekdX?=
 =?utf-8?Q?GKeY=3D?=
Content-Type: text/plain; charset="utf-8"
Content-ID: <D2E73B735110E143BAB9F078E152E8C0@namprd12.prod.outlook.com>
Content-Transfer-Encoding: base64
Precedence: bulk
X-Mailing-List: linux-input@vger.kernel.org
List-Id: <linux-input.vger.kernel.org>
List-Subscribe: <mailto:linux-input+subscribe@vger.kernel.org>
List-Unsubscribe: <mailto:linux-input+unsubscribe@vger.kernel.org>
MIME-Version: 1.0
X-OriginatorOrg: amd.com
X-MS-Exchange-CrossTenant-AuthAs: Internal
X-MS-Exchange-CrossTenant-AuthSource: MN0PR12MB6101.namprd12.prod.outlook.com
X-MS-Exchange-CrossTenant-Network-Message-Id: 03a38bb2-decd-48e3-d336-08ddb266889e
X-MS-Exchange-CrossTenant-originalarrivaltime: 23 Jun 2025 14:59:20.4693
 (UTC)
X-MS-Exchange-CrossTenant-fromentityheader: Hosted
X-MS-Exchange-CrossTenant-id: 3dd8961f-e488-4e60-8e11-a82d994e183d
X-MS-Exchange-CrossTenant-mailboxtype: HOSTED
X-MS-Exchange-CrossTenant-userprincipalname: vHTRPNfo1ttF+jVsrDrmB5Zve5wrqeNSd8DD+CMMp7COwOwwsjQ54aCzBmvYGeQbw0ro3umMVAHOMsFok8NX2A==
X-MS-Exchange-Transport-CrossTenantHeadersStamped: CH8PR12MB9766

T24gNi8yMy8yNSAzOjU0IEFNLCBOYXRpa2FyLCBCYXNhdmFyYWogd3JvdGU6DQo+IEFkZCBjaGFu
Z2VzIHRvIGVuYWJsZSBvcGVyYXRpbmcgbW9kZXMgaW4gdGhlIGRyaXZlciB0byBhbGxvdyB0aGUg
RlcgdG8NCj4gYWN0aXZhdGUgYW5kIHJldHJpZXZlIGRhdGEgZnJvbSByZWxldmFudCBzZW5zb3Jz
LiBUaGlzIGVuYWJsZXMgdGhlIEZXIHRvDQo+IHRha2UgbmVjZXNzYXJ5IGFjdGlvbnMgYmFzZWQg
b24gdGhlIG9wZXJhdGluZyBtb2Rlcy4NCj4gDQo+IFRlc3RlZC1ieTogRXJpYyBOYWltIDxkbmFp
bUBjYWNoeW9zLm9yZz4NCj4gQ28tZGV2ZWxvcGVkLWJ5OiBBa3NoYXRhIE11a3VuZFNoZXR0eSA8
YWtzaGF0YS5tdWt1bmRzaGV0dHlAYW1kLmNvbT4NCj4gU2lnbmVkLW9mZi1ieTogQWtzaGF0YSBN
dWt1bmRTaGV0dHkgPGFrc2hhdGEubXVrdW5kc2hldHR5QGFtZC5jb20+DQo+IFNpZ25lZC1vZmYt
Ynk6IEJhc2F2YXJhaiBOYXRpa2FyIDxCYXNhdmFyYWouTmF0aWthckBhbWQuY29tPg0KUmV2aWV3
ZWQtYnk6IE1hcmlvIExpbW9uY2llbGxvIDxtYXJpby5saW1vbmNpZWxsb0BhbWQuY29tPg0KDQo+
IC0tLQ0KPiB2MjoNCj4gKiBJbmNyZWFzZWQgbWlzc2luZyBkZXZpY2UgY291bnQuDQo+IC0tLQ0K
PiAgIGRyaXZlcnMvaGlkL2FtZC1zZmgtaGlkL2FtZF9zZmhfY2xpZW50LmMgfCAyMyArKysrKysr
KysrKysrKysrKysrKysrKw0KPiAgIGRyaXZlcnMvaGlkL2FtZC1zZmgtaGlkL2FtZF9zZmhfaGlk
LmggICAgfCAgMiArLQ0KPiAgIGRyaXZlcnMvaGlkL2FtZC1zZmgtaGlkL2FtZF9zZmhfcGNpZS5j
ICAgfCAgNCArKysrDQo+ICAgZHJpdmVycy9oaWQvYW1kLXNmaC1oaWQvYW1kX3NmaF9wY2llLmgg
ICB8ICAxICsNCj4gICA0IGZpbGVzIGNoYW5nZWQsIDI5IGluc2VydGlvbnMoKyksIDEgZGVsZXRp
b24oLSkNCj4gDQo+IGRpZmYgLS1naXQgYS9kcml2ZXJzL2hpZC9hbWQtc2ZoLWhpZC9hbWRfc2Zo
X2NsaWVudC5jIGIvZHJpdmVycy9oaWQvYW1kLXNmaC1oaWQvYW1kX3NmaF9jbGllbnQuYw0KPiBp
bmRleCAzNDM4ZDM5MjkyMGYuLjBmMmNiYWUzOWIyYiAxMDA2NDQNCj4gLS0tIGEvZHJpdmVycy9o
aWQvYW1kLXNmaC1oaWQvYW1kX3NmaF9jbGllbnQuYw0KPiArKysgYi9kcml2ZXJzL2hpZC9hbWQt
c2ZoLWhpZC9hbWRfc2ZoX2NsaWVudC5jDQo+IEBAIC0xNDYsNiArMTQ2LDggQEAgc3RhdGljIGNv
bnN0IGNoYXIgKmdldF9zZW5zb3JfbmFtZShpbnQgaWR4KQ0KPiAgIAkJcmV0dXJuICJneXJvc2Nv
cGUiOw0KPiAgIAljYXNlIG1hZ19pZHg6DQo+ICAgCQlyZXR1cm4gIm1hZ25ldG9tZXRlciI7DQo+
ICsJY2FzZSBvcF9pZHg6DQo+ICsJCXJldHVybiAib3BlcmF0aW5nLW1vZGUiOw0KPiAgIAljYXNl
IGFsc19pZHg6DQo+ICAgCWNhc2UgQUNTX0lEWDogLyogYW1iaWVudCBjb2xvciBzZW5zb3IgKi8N
Cj4gICAJCXJldHVybiAiQUxTIjsNCj4gQEAgLTI0Myw2ICsyNDUsMjAgQEAgaW50IGFtZF9zZmhf
aGlkX2NsaWVudF9pbml0KHN0cnVjdCBhbWRfbXAyX2RldiAqcHJpdmRhdGEpDQo+ICAgCQkJcmMg
PSAtRU5PTUVNOw0KPiAgIAkJCWdvdG8gY2xlYW51cDsNCj4gICAJCX0NCj4gKw0KPiArCQlpZiAo
Y2xfZGF0YS0+c2Vuc29yX2lkeFtpXSA9PSBvcF9pZHgpIHsNCj4gKwkJCWluZm8ucGVyaW9kID0g
QU1EX1NGSF9JRExFX0xPT1A7DQo+ICsJCQlpbmZvLnNlbnNvcl9pZHggPSBjbF9kYXRhLT5zZW5z
b3JfaWR4W2ldOw0KPiArCQkJaW5mby5kbWFfYWRkcmVzcyA9IGNsX2RhdGEtPnNlbnNvcl9kbWFf
YWRkcltpXTsNCj4gKwkJCW1wMl9vcHMtPnN0YXJ0KHByaXZkYXRhLCBpbmZvKTsNCj4gKwkJCWNs
X2RhdGEtPnNlbnNvcl9zdHNbaV0gPSBhbWRfc2ZoX3dhaXRfZm9yX3Jlc3BvbnNlKHByaXZkYXRh
LA0KPiArCQkJCQkJCQkJICAgY2xfZGF0YS0+c2Vuc29yX2lkeFtpXSwNCj4gKwkJCQkJCQkJCSAg
IFNFTlNPUl9FTkFCTEVEKTsNCj4gKwkJCWlmIChjbF9kYXRhLT5zZW5zb3Jfc3RzW2ldID09IFNF
TlNPUl9FTkFCTEVEKQ0KPiArCQkJCWNsX2RhdGEtPmlzX2FueV9zZW5zb3JfZW5hYmxlZCA9IHRy
dWU7DQo+ICsJCQljb250aW51ZTsNCj4gKwkJfQ0KPiArDQo+ICAgCQljbF9kYXRhLT5zZW5zb3Jf
c3RzW2ldID0gU0VOU09SX0RJU0FCTEVEOw0KPiAgIAkJY2xfZGF0YS0+c2Vuc29yX3JlcXVlc3Rl
ZF9jbnRbaV0gPSAwOw0KPiAgIAkJY2xfZGF0YS0+Y3VyX2hpZF9kZXYgPSBpOw0KPiBAQCAtMzAz
LDYgKzMxOSwxMyBAQCBpbnQgYW1kX3NmaF9oaWRfY2xpZW50X2luaXQoc3RydWN0IGFtZF9tcDJf
ZGV2ICpwcml2ZGF0YSkNCj4gICANCj4gICAJZm9yIChpID0gMDsgaSA8IGNsX2RhdGEtPm51bV9o
aWRfZGV2aWNlczsgaSsrKSB7DQo+ICAgCQljbF9kYXRhLT5jdXJfaGlkX2RldiA9IGk7DQo+ICsJ
CWlmIChjbF9kYXRhLT5zZW5zb3JfaWR4W2ldID09IG9wX2lkeCkgew0KPiArCQkJZGV2X2RiZyhk
ZXYsICJzaWQgMHgleCAoJXMpIHN0YXR1cyAweCV4XG4iLA0KPiArCQkJCWNsX2RhdGEtPnNlbnNv
cl9pZHhbaV0sIGdldF9zZW5zb3JfbmFtZShjbF9kYXRhLT5zZW5zb3JfaWR4W2ldKSwNCj4gKwkJ
CQljbF9kYXRhLT5zZW5zb3Jfc3RzW2ldKTsNCj4gKwkJCWNvbnRpbnVlOw0KPiArCQl9DQo+ICsN
Cj4gICAJCWlmIChjbF9kYXRhLT5zZW5zb3Jfc3RzW2ldID09IFNFTlNPUl9FTkFCTEVEKSB7DQo+
ICAgCQkJcmMgPSBhbWR0cF9oaWRfcHJvYmUoaSwgY2xfZGF0YSk7DQo+ICAgCQkJaWYgKHJjKQ0K
PiBkaWZmIC0tZ2l0IGEvZHJpdmVycy9oaWQvYW1kLXNmaC1oaWQvYW1kX3NmaF9oaWQuaCBiL2Ry
aXZlcnMvaGlkL2FtZC1zZmgtaGlkL2FtZF9zZmhfaGlkLmgNCj4gaW5kZXggMWM5MWJlOGRhZWRk
Li43NDUyYjAzMDI5NTMgMTAwNjQ0DQo+IC0tLSBhL2RyaXZlcnMvaGlkL2FtZC1zZmgtaGlkL2Ft
ZF9zZmhfaGlkLmgNCj4gKysrIGIvZHJpdmVycy9oaWQvYW1kLXNmaC1oaWQvYW1kX3NmaF9oaWQu
aA0KPiBAQCAtMTEsNyArMTEsNyBAQA0KPiAgICNpZm5kZWYgQU1EU0ZIX0hJRF9IDQo+ICAgI2Rl
ZmluZSBBTURTRkhfSElEX0gNCj4gICANCj4gLSNkZWZpbmUgTUFYX0hJRF9ERVZJQ0VTCQk2DQo+
ICsjZGVmaW5lIE1BWF9ISURfREVWSUNFUwkJNw0KPiAgICNkZWZpbmUgQU1EX1NGSF9ISURfVkVO
RE9SCTB4MTAyMg0KPiAgICNkZWZpbmUgQU1EX1NGSF9ISURfUFJPRFVDVAkweDAwMDENCj4gICAN
Cj4gZGlmZiAtLWdpdCBhL2RyaXZlcnMvaGlkL2FtZC1zZmgtaGlkL2FtZF9zZmhfcGNpZS5jIGIv
ZHJpdmVycy9oaWQvYW1kLXNmaC1oaWQvYW1kX3NmaF9wY2llLmMNCj4gaW5kZXggMWMxZmQ2MzMz
MGM5Li4yOTgzYWY5Njk1NzkgMTAwNjQ0DQo+IC0tLSBhL2RyaXZlcnMvaGlkL2FtZC1zZmgtaGlk
L2FtZF9zZmhfcGNpZS5jDQo+ICsrKyBiL2RyaXZlcnMvaGlkL2FtZC1zZmgtaGlkL2FtZF9zZmhf
cGNpZS5jDQo+IEBAIC0yOSw2ICsyOSw3IEBADQo+ICAgI2RlZmluZSBBQ0VMX0VOCQlCSVQoMCkN
Cj4gICAjZGVmaW5lIEdZUk9fRU4JCUJJVCgxKQ0KPiAgICNkZWZpbmUgTUFHTk9fRU4JQklUKDIp
DQo+ICsjZGVmaW5lIE9QX0VOCQlCSVQoMTUpDQo+ICAgI2RlZmluZSBIUERfRU4JCUJJVCgxNikN
Cj4gICAjZGVmaW5lIEFMU19FTgkJQklUKDE5KQ0KPiAgICNkZWZpbmUgQUNTX0VOCQlCSVQoMjIp
DQo+IEBAIC0yMzIsNiArMjMzLDkgQEAgaW50IGFtZF9tcDJfZ2V0X3NlbnNvcl9udW0oc3RydWN0
IGFtZF9tcDJfZGV2ICpwcml2ZGF0YSwgdTggKnNlbnNvcl9pZCkNCj4gICAJaWYgKE1BR05PX0VO
ICYgYWN0aXZlc3RhdHVzKQ0KPiAgIAkJc2Vuc29yX2lkW251bV9vZl9zZW5zb3JzKytdID0gbWFn
X2lkeDsNCj4gICANCj4gKwlpZiAoT1BfRU4gJiBhY3RpdmVzdGF0dXMpDQo+ICsJCXNlbnNvcl9p
ZFtudW1fb2Zfc2Vuc29ycysrXSA9IG9wX2lkeDsNCj4gKw0KPiAgIAlpZiAoQUxTX0VOICYgYWN0
aXZlc3RhdHVzKQ0KPiAgIAkJc2Vuc29yX2lkW251bV9vZl9zZW5zb3JzKytdID0gYWxzX2lkeDsN
Cj4gICANCj4gZGlmZiAtLWdpdCBhL2RyaXZlcnMvaGlkL2FtZC1zZmgtaGlkL2FtZF9zZmhfcGNp
ZS5oIGIvZHJpdmVycy9oaWQvYW1kLXNmaC1oaWQvYW1kX3NmaF9wY2llLmgNCj4gaW5kZXggMDVl
NDAwYTRhODNlLi4yZWI2MWY0ZTg0MzQgMTAwNjQ0DQo+IC0tLSBhL2RyaXZlcnMvaGlkL2FtZC1z
ZmgtaGlkL2FtZF9zZmhfcGNpZS5oDQo+ICsrKyBiL2RyaXZlcnMvaGlkL2FtZC1zZmgtaGlkL2Ft
ZF9zZmhfcGNpZS5oDQo+IEBAIC03OSw2ICs3OSw3IEBAIGVudW0gc2Vuc29yX2lkeCB7DQo+ICAg
CWFjY2VsX2lkeCA9IDAsDQo+ICAgCWd5cm9faWR4ID0gMSwNCj4gICAJbWFnX2lkeCA9IDIsDQo+
ICsJb3BfaWR4ID0gMTUsDQo+ICAgCWFsc19pZHggPSAxOQ0KPiAgIH07DQo+ICAgDQoNCg==

From mboxrd@z Thu Jan  1 00:00:00 1970
Received: from smtp.kernel.org (aws-us-west-2-korg-mail-1.web.codeaurora.org [10.30.226.201])
	(using TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384 (256/256 bits))
	(No client certificate requested)
	by smtp.subspace.kernel.org (Postfix) with ESMTPS id 2D5CD221F37
	for <linux-input@vger.kernel.org>; Thu,  3 Jul 2025 07:36:14 +0000 (UTC)
Authentication-Results: smtp.subspace.kernel.org; arc=none smtp.client-ip=10.30.226.201
ARC-Seal:i=1; a=rsa-sha256; d=subspace.kernel.org; s=arc-20240116;
	t=1751528174; cv=none; b=EKORI8zEJ2yaVo8LNwxzRdlVLDaTMvy/E5EXaBF/pRoN5GlQzOs5EcuGTVQbPLseFnAywc8mqj6nThBI5LM0F7/eDxVMxoLuZZxDAqOmHjP5EuhhvxRjDdk8KcFQ5r/9JYO4hJmxj7OKoH8cnc/tky0PcmZlaspHFrGMIVisJQU=
ARC-Message-Signature:i=1; a=rsa-sha256; d=subspace.kernel.org;
	s=arc-20240116; t=1751528174; c=relaxed/simple;
	bh=imYNRsq3vEhiy1/Gv4xLEhV5iOWP+vQDFSqpbRx/sfo=;
	h=Date:From:To:cc:Subject:In-Reply-To:Message-ID:References:
	 MIME-Version:Content-Type; b=gGXbg1ioylT9FIoXpT2HBJgS+gH494QoCvs/hSG9dJ7+aBGlEoie0Yvm9xC/ghFpkm4x/UlStzSXQbmsDu17JEzNge20gcVjUuxZstWUYxT4x0MYQaY2QgdG01uZiy8gg5yKfiGKjxvI7Y0OwzUslyQaAUdCHeogrZrzKKC0zJg=
ARC-Authentication-Results:i=1; smtp.subspace.kernel.org; dkim=pass (2048-bit key) header.d=kernel.org header.i=@kernel.org header.b=d8yRTtKT; arc=none smtp.client-ip=10.30.226.201
Authentication-Results: smtp.subspace.kernel.org;
	dkim=pass (2048-bit key) header.d=kernel.org header.i=@kernel.org header.b="d8yRTtKT"
Received: by smtp.kernel.org (Postfix) with ESMTPSA id 90826C4CEE3;
	Thu,  3 Jul 2025 07:36:13 +0000 (UTC)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/simple; d=kernel.org;
	s=k20201202; t=1751528174;
	bh=imYNRsq3vEhiy1/Gv4xLEhV5iOWP+vQDFSqpbRx/sfo=;
	h=Date:From:To:cc:Subject:In-Reply-To:References:From;
	b=d8yRTtKT5buqvcnJ7jhXKSVw135A9R9O79fXMVHm+MGRH5daIkjQBi5BR++MHav9/
	 s+GLXPs549qwcDkO46ScLpbuyxKQmtSFqwwAa4aFOhzUhBjpclfuUqIS0nlno5Z8Fr
	 TKCXgDTmQA37WZkZdm73CfeJYjmkvvguRfb3iU7RHmctAHsNLvO+c224eGRGJNRZKB
	 RqNhYDxmw1bn4R/OKnqWb4lMXHcUXdmW/XoO1NJQUsxumYXEh0WptbJ54A4CF786p1
	 CkoTMjVmZSZLWNhSHnvqG5mVUH4bHIJcxCH5QZZXBIVU5rowqO5s07jzuCP3gvDzcV
	 vuwOuOeibo+Pw==
Date: Thu, 3 Jul 2025 09:36:11 +0200 (CEST)
From: Jiri Kosina <jikos@kernel.org>
To: Basavaraj Natikar <Basavaraj.Natikar@amd.com>
cc: bentiss@kernel.org, Mario.Limonciello@amd.com, linux-input@vger.kernel.org, 
    Eric Naim <dnaim@cachyos.org>, 
    Akshata MukundShetty <akshata.mukundshetty@amd.com>
Subject: Re: [PATCH v2] HID: amd_sfh: Enable operating mode
In-Reply-To: <20250623085453.2267034-1-Basavaraj.Natikar@amd.com>
Message-ID: <1o3n7op6-775o-713n-2n9n-n6rs300o4825@xreary.bet>
References: <20250623085453.2267034-1-Basavaraj.Natikar@amd.com>
Precedence: bulk
X-Mailing-List: linux-input@vger.kernel.org
List-Id: <linux-input.vger.kernel.org>
List-Subscribe: <mailto:linux-input+subscribe@vger.kernel.org>
List-Unsubscribe: <mailto:linux-input+unsubscribe@vger.kernel.org>
MIME-Version: 1.0
Content-Type: text/plain; charset=US-ASCII

On Mon, 23 Jun 2025, Basavaraj Natikar wrote:

> Add changes to enable operating modes in the driver to allow the FW to
> activate and retrieve data from relevant sensors. This enables the FW to
> take necessary actions based on the operating modes.
> 
> Tested-by: Eric Naim <dnaim@cachyos.org>
> Co-developed-by: Akshata MukundShetty <akshata.mukundshetty@amd.com>
> Signed-off-by: Akshata MukundShetty <akshata.mukundshetty@amd.com>
> Signed-off-by: Basavaraj Natikar <Basavaraj.Natikar@amd.com>

Applied to hid.git#for-6.17/amd-sfh. Thanks,

-- 
Jiri Kosina
SUSE Labs


From mboxrd@z Thu Jan  1 00:00:00 1970
Received: from mail-ej1-f42.google.com (mail-ej1-f42.google.com [209.85.218.42])
	(using TLSv1.2 with cipher ECDHE-RSA-AES128-GCM-SHA256 (128/128 bits))
	(No client certificate requested)
	by smtp.subspace.kernel.org (Postfix) with ESMTPS id 250B825393C
	for <linux-input@vger.kernel.org>; Thu,  7 Aug 2025 12:43:00 +0000 (UTC)
Authentication-Results: smtp.subspace.kernel.org; arc=none smtp.client-ip=209.85.218.42
ARC-Seal:i=1; a=rsa-sha256; d=subspace.kernel.org; s=arc-20240116;
	t=1754570583; cv=none; b=W8HTdT2J3YMlYB/5z75ZtIXYRychILhdMNhp3SjjCSLx75BdS7HfONWwxPW3f0Aery4Q7RHHG5bkTk2ETlmLZpPNwghsw7Av7J76HvLqXyl8uzu2amXM5RiLKt5i6yjp0vafDf2hgb+eiTq8K29EWO9yqUQzVVlMhIBfP3cGshI=
ARC-Message-Signature:i=1; a=rsa-sha256; d=subspace.kernel.org;
	s=arc-20240116; t=1754570583; c=relaxed/simple;
	bh=3zgP7iR6K7q9sLZ8DSbuc508pNd9u2UBGuvPUziUXxQ=;
	h=From:To:Cc:Subject:Date:Message-ID:In-Reply-To:References:
	 MIME-Version:Content-Type; b=W8sVvPB3KvRWKYn3Crfmlmbk3xeOKvHHZr+a7jzCGVHrJ5NO5gZEFfNvcNRL2pqfZQBSTUAGEoG1eYEwzzl6SQREuTykx1C0bjG51DYcbX7Jn6Tla8JtTfwoKuuXWzi3nfUjnxwORgwrtGTYSRqVoqIgSXhVHoyMAfqWVoJe8I8=
ARC-Authentication-Results:i=1; smtp.subspace.kernel.org; dmarc=pass (p=none dis=none) header.from=gmail.com; spf=pass smtp.mailfrom=gmail.com; dkim=pass (2048-bit key) header.d=gmail.com header.i=@gmail.com header.b=cpSakPD5; arc=none smtp.client-ip=209.85.218.42
Authentication-Results: smtp.subspace.kernel.org; dmarc=pass (p=none dis=none) header.from=gmail.com
Authentication-Results: smtp.subspace.kernel.org; spf=pass smtp.mailfrom=gmail.com
Authentication-Results: smtp.subspace.kernel.org;
	dkim=pass (2048-bit key) header.d=gmail.com header.i=@gmail.com header.b="cpSakPD5"
Received: by mail-ej1-f42.google.com with SMTP id a640c23a62f3a-ae9c2754a00so192024566b.2
        for <linux-input@vger.kernel.org>; Thu, 07 Aug 2025 05:43:00 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=gmail.com; s=20230601; t=1754570579; x=1755175379; darn=vger.kernel.org;
        h=content-transfer-encoding:mime-version:references:in-reply-to
         :message-id:date:subject:cc:to:from:from:to:cc:subject:date
         :message-id:reply-to;
        bh=p+FEF9xNUl87NULgl2gpCQyvIOR29nQAQJmVR0Lxk4Y=;
        b=cpSakPD5XYoxKp3XB7LIUlI2Aa0RDggmenZqs9HeQo3b2V/Znc5jP37xF7iSoU5h2C
         SKulhOkBN4iLpto0UsJhyrv3Ri3M1E26YIr9hhRPFminYmD7dt9UoCjub9QfSGpUyKbj
         DYqbseXFyl0UzJM9anbexpQIbY1/7TCJgWS1KNOfNwEbO6u7qD7A2bFTW2M8wUMLa5iC
         jjR/T1rSiZdw5MK9cwg4xroryEFzROihvC0ixQobhAMqkJgu6SnQr1lr+PJrgpiF3yub
         rqxBjnNe8HDNsvJG2ul1F7De0+gi0ylmQMuGKKFdf9o0dOwCX7u+crD92lYzOTL1R3JM
         1nLw==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20230601; t=1754570579; x=1755175379;
        h=content-transfer-encoding:mime-version:references:in-reply-to
         :message-id:date:subject:cc:to:from:x-gm-message-state:from:to:cc
         :subject:date:message-id:reply-to;
        bh=p+FEF9xNUl87NULgl2gpCQyvIOR29nQAQJmVR0Lxk4Y=;
        b=WRqTOwwnePrtEOkPbbRB1KFdTY7Exfke5eQJ6Oi5IeUMaGp7dO3/sfpKcQ1uCUPlgG
         CMQuOT1EABRKvH9sQI9tcNk2eqIHRb39To14yMpasw0EJMPESw1y/JGJpI2M3pRsDGxw
         khk/cX9vfcmdVxPncNf7JwPpGS80qIUw1Ac1oeudn0uXZM/EmunFZZvJ9cqx1kyGuzl3
         D0PmTD+lweX036faoFg0r2DLHSCwRBxz2i384e40kW/0Y+qikQY/O9aEsm/EyfUXsZpN
         EEHs4gNieBzAcfyGRxAiHemyHvCyhd9chQ/nAPrG+JYCBLBL1QLyLVeVtiebEcW8ynLq
         QpJw==
X-Forwarded-Encrypted: i=1; AJvYcCVSd8uSFQJ+4HrQRDIapmdZn26QxXGNNw9hcMY7vIjKG2bIUiNAEtjNzLgwlw8u/ovmvk3IZFGEL6BNuw==@vger.kernel.org
X-Gm-Message-State: AOJu0YygT+XABtgCCQKmf8Tas09L8I4HTq1B3awCInuR9A912G7MvgX+
	wVLlU8jvC0XV1gmpYArYaS1OYYfJS0h9WdyTcIjDmkoTvW1L5DbWg4Tq
X-Gm-Gg: ASbGnctGsT6TSl/xSIE2rUEfGk167gsfpzNMqv7WOdYArXD/1CctEGa5yOZrOi0U9//
	OA8Tc1AmlxBJ21Ebb9TwvrsJ7NEIQ0NkIi2EcqGF3LIoYtkVRfCLjVJjj1LK0+8roATYLc5lUBb
	vdFzfZSWEYKy/6l4M/d3xvcWMTMlmB+F6N/YDjFtCcn8tHN9H4XdDj+GykE1h/BNxth2KHNfzkR
	4MFM1CEwypsGtX6HiaBUUEtmLBpFVY4pvzD1OiEAgiboAlgiVwZcUdvGTPUmZlXmuoPox+DMV3H
	3J9JkQ3/PwyNz78Xekv2i+QyRliRBChTCSnOvQXHKbCjGwMsGDccp4adC/tnDy+aFWJh8nqpfOu
	SAfj7cqvBikU955MFQIDZf/fpYw==
X-Google-Smtp-Source: AGHT+IFd38Ak1KsXIH8QDdjHIZb+w+PSHX7kL/etdHGM3/kvJMbPEj2e9COm/bqyVP8/s6+LH2Fllw==
X-Received: by 2002:a17:906:c10b:b0:af9:3116:e0f6 with SMTP id a640c23a62f3a-af9904baf22mr671444066b.53.1754570579083;
        Thu, 07 Aug 2025 05:42:59 -0700 (PDT)
Received: from fedora.localnet ([79.127.145.93])
        by smtp.gmail.com with ESMTPSA id a640c23a62f3a-af92a2ec9cbsm1189034766b.79.2025.08.07.05.42.57
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Thu, 07 Aug 2025 05:42:58 -0700 (PDT)
From: butmonkeh@gmail.com
To: jikos@kernel.org, bentiss@kernel.org,
 Basavaraj Natikar <Basavaraj.Natikar@amd.com>
Cc: Mario.Limonciello@amd.com, linux-input@vger.kernel.org,
 Basavaraj Natikar <Basavaraj.Natikar@amd.com>, Eric Naim <dnaim@cachyos.org>,
 Akshata MukundShetty <akshata.mukundshetty@amd.com>
Subject: Re: [PATCH v2] HID: amd_sfh: Enable operating mode
Date: Thu, 07 Aug 2025 13:42:56 +0100
Message-ID: <5275875.OV4Wx5bFTl@fedora>
In-Reply-To: <20250623085453.2267034-1-Basavaraj.Natikar@amd.com>
References: <20250623085453.2267034-1-Basavaraj.Natikar@amd.com>
Precedence: bulk
X-Mailing-List: linux-input@vger.kernel.org
List-Id: <linux-input.vger.kernel.org>
List-Subscribe: <mailto:linux-input+subscribe@vger.kernel.org>
List-Unsubscribe: <mailto:linux-input+unsubscribe@vger.kernel.org>
MIME-Version: 1.0
Content-Transfer-Encoding: 7Bit
Content-Type: text/plain; charset="utf-8"

On Monday, 23 June 2025 09:54:53 British Summer Time Basavaraj Natikar wrote:
> Add changes to enable operating modes in the driver to allow the FW to
> activate and retrieve data from relevant sensors. This enables the FW to
> take necessary actions based on the operating modes.
> 
> Tested-by: Eric Naim <dnaim@cachyos.org>
> Co-developed-by: Akshata MukundShetty <akshata.mukundshetty@amd.com>
> Signed-off-by: Akshata MukundShetty <akshata.mukundshetty@amd.com>
> Signed-off-by: Basavaraj Natikar <Basavaraj.Natikar@amd.com>
> ---
> v2:
> * Increased missing device count.
> ---
>  drivers/hid/amd-sfh-hid/amd_sfh_client.c | 23 +++++++++++++++++++++++
>  drivers/hid/amd-sfh-hid/amd_sfh_hid.h    |  2 +-
>  drivers/hid/amd-sfh-hid/amd_sfh_pcie.c   |  4 ++++
>  drivers/hid/amd-sfh-hid/amd_sfh_pcie.h   |  1 +
>  4 files changed, 29 insertions(+), 1 deletion(-)
> 
> diff --git a/drivers/hid/amd-sfh-hid/amd_sfh_client.c
> b/drivers/hid/amd-sfh-hid/amd_sfh_client.c index 3438d392920f..0f2cbae39b2b
> 100644
> --- a/drivers/hid/amd-sfh-hid/amd_sfh_client.c
> +++ b/drivers/hid/amd-sfh-hid/amd_sfh_client.c
> @@ -146,6 +146,8 @@ static const char *get_sensor_name(int idx)
>  		return "gyroscope";
>  	case mag_idx:
>  		return "magnetometer";
> +	case op_idx:
> +		return "operating-mode";
>  	case als_idx:
>  	case ACS_IDX: /* ambient color sensor */
>  		return "ALS";
> @@ -243,6 +245,20 @@ int amd_sfh_hid_client_init(struct amd_mp2_dev
> *privdata) rc = -ENOMEM;
>  			goto cleanup;
>  		}
> +
> +		if (cl_data->sensor_idx[i] == op_idx) {
> +			info.period = AMD_SFH_IDLE_LOOP;
> +			info.sensor_idx = cl_data->sensor_idx[i];
> +			info.dma_address = cl_data->sensor_dma_addr[i];
> +			mp2_ops->start(privdata, info);
> +			cl_data->sensor_sts[i] = 
amd_sfh_wait_for_response(privdata,
> +								
	   cl_data->sensor_idx[i],
> +								
	   SENSOR_ENABLED);
> +			if (cl_data->sensor_sts[i] == SENSOR_ENABLED)
> +				cl_data->is_any_sensor_enabled = true;
> +			continue;
> +		}
> +
>  		cl_data->sensor_sts[i] = SENSOR_DISABLED;
>  		cl_data->sensor_requested_cnt[i] = 0;
>  		cl_data->cur_hid_dev = i;
> @@ -303,6 +319,13 @@ int amd_sfh_hid_client_init(struct amd_mp2_dev
> *privdata)
> 
>  	for (i = 0; i < cl_data->num_hid_devices; i++) {
>  		cl_data->cur_hid_dev = i;
> +		if (cl_data->sensor_idx[i] == op_idx) {
> +			dev_dbg(dev, "sid 0x%x (%s) status 0x%x\n",
> +				cl_data->sensor_idx[i], 
get_sensor_name(cl_data->sensor_idx[i]),
> +				cl_data->sensor_sts[i]);
> +			continue;
> +		}
> +
>  		if (cl_data->sensor_sts[i] == SENSOR_ENABLED) {
>  			rc = amdtp_hid_probe(i, cl_data);
>  			if (rc)
> diff --git a/drivers/hid/amd-sfh-hid/amd_sfh_hid.h
> b/drivers/hid/amd-sfh-hid/amd_sfh_hid.h index 1c91be8daedd..7452b0302953
> 100644
> --- a/drivers/hid/amd-sfh-hid/amd_sfh_hid.h
> +++ b/drivers/hid/amd-sfh-hid/amd_sfh_hid.h
> @@ -11,7 +11,7 @@
>  #ifndef AMDSFH_HID_H
>  #define AMDSFH_HID_H
> 
> -#define MAX_HID_DEVICES		6
> +#define MAX_HID_DEVICES		7
>  #define AMD_SFH_HID_VENDOR	0x1022
>  #define AMD_SFH_HID_PRODUCT	0x0001
> 
> diff --git a/drivers/hid/amd-sfh-hid/amd_sfh_pcie.c
> b/drivers/hid/amd-sfh-hid/amd_sfh_pcie.c index 1c1fd63330c9..2983af969579
> 100644
> --- a/drivers/hid/amd-sfh-hid/amd_sfh_pcie.c
> +++ b/drivers/hid/amd-sfh-hid/amd_sfh_pcie.c
> @@ -29,6 +29,7 @@
>  #define ACEL_EN		BIT(0)
>  #define GYRO_EN		BIT(1)
>  #define MAGNO_EN	BIT(2)
> +#define OP_EN		BIT(15)
>  #define HPD_EN		BIT(16)
>  #define ALS_EN		BIT(19)
>  #define ACS_EN		BIT(22)
> @@ -232,6 +233,9 @@ int amd_mp2_get_sensor_num(struct amd_mp2_dev *privdata,
> u8 *sensor_id) if (MAGNO_EN & activestatus)
>  		sensor_id[num_of_sensors++] = mag_idx;
> 
> +	if (OP_EN & activestatus)
> +		sensor_id[num_of_sensors++] = op_idx;
> +
>  	if (ALS_EN & activestatus)
>  		sensor_id[num_of_sensors++] = als_idx;
> 
> diff --git a/drivers/hid/amd-sfh-hid/amd_sfh_pcie.h
> b/drivers/hid/amd-sfh-hid/amd_sfh_pcie.h index 05e400a4a83e..2eb61f4e8434
> 100644
> --- a/drivers/hid/amd-sfh-hid/amd_sfh_pcie.h
> +++ b/drivers/hid/amd-sfh-hid/amd_sfh_pcie.h
> @@ -79,6 +79,7 @@ enum sensor_idx {
>  	accel_idx = 0,
>  	gyro_idx = 1,
>  	mag_idx = 2,
> +	op_idx = 15,
>  	als_idx = 19
>  };

Tested-by: Marc H (butmonkeh@gmail.com) Asus ROG Flow X13 (GV301) - working



